### ISSUE TITLE
`systemctl enable`, `systemctl add-wants` do not work for template units with no instance specified

### ISSUE THREAD

**Issue by intelfx** [systemd/systemd#3473]
### Submission type
- [X] Bug report
- [ ] Request for enhancement (RFE)

This is a resubmit of the [fd.o issue 83110](https://bugs.freedesktop.org/show_bug.cgi?id=83110).
### systemd version the issue has been seen with

v230
### Used distribution

Arch Linux
### In case of bug report: Steps to reproduce the problem

```
$ systemctl --user cat test@.target test1@.service
# /home/operator/.config/systemd/user/test@.target
[Unit]
Description=Test target, instance %i

# /home/operator/.config/systemd/user/test1@.service
[Unit]
Description=Test service 1, instance %i

[Service]
Type=oneshot
ExecStart=/bin/echo "test1@%i"

[Install]
WantedBy=test@.target

$ systemctl --user enable test1@
The unit files have no installation config (WantedBy, RequiredBy, Also, Alias
settings in the [Install] section, and DefaultInstance for template units).
This means they are not meant to be enabled using systemctl.
Possible reasons for having this kind of units are:
1) A unit may be statically enabled by being symlinked from another unit's
   .wants/ or .requires/ directory.
2) A unit's purpose may be to act as a helper for some other unit which has
   a requirement dependency on it.
3) A unit may be started when needed via activation (socket, path, timer,
   D-Bus, udev, scripted systemctl call, ...).
4) In case of template units, the unit is meant to be enabled with some
   instance name specified.

$ systemctl --user add-wants test@ test1@                 
<no output>

```
### In case of bug report: Expected behaviour you didn't see

```
$ LC_ALL=C ls -la ~/.config/systemd/user/test@.target.wants
total 4.0K
drwx------+ 1 operator users   28 Jun  9 05:16 .
drwxr-xr-x+ 1 operator users 1.3K Jun  9 05:16 ..
lrwxrwxrwx  1 operator users   50 Jun  9 05:16 test1@.service -> /home/operator/.config/systemd/user/test1@.service
```
### In case of bug report: Unexpected behaviour you saw

```

$ LC_ALL=C ls -la ~/.config/systemd/user/test@.target.wants
ls: cannot access '/home/operator/.config/systemd/user/test@.target.wants': No such file or directory
```

**Comment by keszybz** [comment]
Still present with current git.

**Comment by poettering** [comment]
Fix waiting in #4992.
